tests/components/monzo/test_sensor.py

```diff
@@ -0,0 +1,142 @@
+
+from homeassistant.components.monzo.const import DOMAIN
+from homeassistant.components.monzo.sensor import (
+    ACC_SENSORS,
+    POT_SENSORS,
+    MonzoSensorEntityDescription,
+)
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.const import STATE_UNAVAILABLE
+from homeassistant.core import HomeAssistant, State
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.entity_registry import EntityRegistry
+
+from . import setup_integration
+from .conftest import TEST_ACCOUNTS, TEST_POTS
+
+from tests.common import MockConfigEntry, async_fire_time_changed
+from tests.typing import ClientSessionGenerator
+
+EXPECTED_VALUE_GETTERS = {
+    "balance": lambda x: x["balance"]["balance"] / 100,
+    "total_balance": lambda x: x["balance"]["total_balance"] / 100,
+    "pot_balance": lambda x: x["balance"] / 100,
+}
+
+
+async def async_get_entity_id(
+    hass: HomeAssistant,
+    acc_id: str,
+    description: MonzoSensorEntityDescription,
+) -> str | None:
+    """Get an entity id for a user's attribute."""
+    entity_registry = er.async_get(hass)
+    unique_id = f"{acc_id}_{description.key}"
+
+    return entity_registry.async_get_entity_id(SENSOR_DOMAIN, DOMAIN, unique_id)
+
+
+def async_assert_state_equals(
+    entity_id: str,
+    state_obj: State,
+    expected: Any,
+    description: MonzoSensorEntityDescription,
+) -> None:
+    """Assert at given state matches what is expected."""
+    assert state_obj, f"Expected entity {entity_id} to exist but it did not"
+
+    assert state_obj.state == str(expected), (
+        f"Expected {expected} but was {state_obj.state} "
+        f"for measure {description.name}, {entity_id}"
+    )
+
+
+@pytest.mark.usefixtures("entity_registry_enabled_by_default")
+async def test_sensor_default_enabled_entities(
+    hass: HomeAssistant,
+    monzo: AsyncMock,
+    polling_config_entry: MockConfigEntry,
+    hass_client_no_auth: ClientSessionGenerator,
+) -> None:
+    """Test entities enabled by default."""
+    await setup_integration(hass, polling_config_entry)
+    entity_registry: EntityRegistry = er.async_get(hass)
+
+    for acc in TEST_ACCOUNTS:
+        for sensor_description in ACC_SENSORS:
+            entity_id = await async_get_entity_id(hass, acc["id"], sensor_description)
+            assert entity_id
+            assert entity_registry.async_is_registered(entity_id)
+
+            state = hass.states.get(entity_id)
+            assert state.state == str(
+                EXPECTED_VALUE_GETTERS[sensor_description.key](acc)
+            )
+
+
+@pytest.mark.usefixtures("entity_registry_enabled_by_default")
+async def test_failure(
+    hass: HomeAssistant,
+    basic_monzo: AsyncMock,
+    polling_config_entry: MockConfigEntry,
+    hass_client_no_auth: ClientSessionGenerator,
+) -> None:
+    """Test entities enabled by default."""
+    with patch(
+        "homeassistant.components.monzo.sensor.MonzoBaseEntity.data",
```

Comment:
Instead of patching the integration, can we patch the Monzo API?

Reply:
I had a bit of trouble trying to do it that way, I couldn't figure out how to make HomeAssistant immediately poll the API

Reply:
you can use the freezer to move the time, that should be enough check [this ](https://github.com/home-assistant/core/blob/dev/tests/components/tedee/test_sensor.py#L56-L59) for an example